django_socio_grpc/services.py

```diff
@@ -95,31 +95,60 @@ def force_evaluation():
             cls.queryset._fetch_all = force_evaluation
 
         class Servicer:
+
+            grpc_async = grpc_settings.GRPC_ASYNC
+
             def __getattr__(self, action):
                 if not hasattr(cls, action):
                     return not_implemented
 
-                def handler(request, context):
-                    # db connection state managed similarly to the wsgi handler
-                    db.reset_queries()
-                    db.close_old_connections()
-                    try:
-                        self = cls(**initkwargs)
-                        self.request = request
-                        self.context = GRPCSocioProxyContext(context, action)
-                        self.action = action
-                        self.before_action()
-                        return getattr(self, action)(self.request, self.context)
-                    except GRPCException as grpc_error:
-                        logger.error(grpc_error)
-                        self.context.abort(
-                            grpc_error.status_code, grpc_error.get_full_details()
-                        )
-                    finally:
-                        db.close_old_connections()
+                if self.grpc_async:
 
-                update_wrapper(handler, getattr(cls, action))
-                return handler
+                    async def async_handler(request, context):
```

Comment:
do you think is possible to extract async handler and handler to a separate file or to put them at the bottom of the file.
Even moving the all Servicer class in an other file to avoid too much complexity in the services files ?

Reply:
Yes Servicer class can be move out this file